src/tree/updater_gpu_hist.cu

```diff
@@ -532,6 +529,13 @@ struct GPUHistMakerDevice {
 
   void ApplySplit(const GPUExpandEntry& candidate, RegTree* p_tree) {
     RegTree& tree = *p_tree;
+
+    // Sanity check - have we created a leaf with no training instances?
+    if (!rabit::IsDistributed() && row_partitioner) {
```

Comment:
Any potential solution if this check occurs?

Reply:
My mistake.  I ran into issues with empty nodes in partitioner `UpdatePosition`, where `left_count` can be 0 due to floating-point errors.  In this `ApplySplit` function, if we check on the left/right child instead of the parent (candidate), then we might run into trouble.